plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt

```diff
@@ -261,23 +264,35 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
     }
 
     private fun isBuildOrBin(filePath: String): Boolean {
-        val regex = Regex("""[/\\](bin|build|node_modules|venv|env|\.idea)[/\\]""", RegexOption.IGNORE_CASE)
+        val regex = Regex("""[/\\](bin|build|node_modules|venv|.venv|env|\.idea)[/\\]""", RegexOption.IGNORE_CASE)
         return regex.find(filePath) != null
     }
 
     private fun collectFiles(): FileCollectionResult {
         val collectedFiles = mutableListOf<String>()
         var currentTotalFileSize = 0L
-        val changeListManager = ChangeListManager.getInstance(project)
         val featureDevSessionContext = FeatureDevSessionContext(project)
-        val allFiles = project.guessProjectDir()?.let {
-            VfsUtil.collectChildrenRecursively(it).filter { child ->
-                !changeListManager.isIgnoredFile(child) &&
-                    !isBuildOrBin(child.path) &&
-                    runBlocking { !featureDevSessionContext.ignoreFile(child, scope) } &&
-                    child.length <= 10 * 1024 * 1024
-            }
-        }.orEmpty()
+        val allFiles: MutableList<VirtualFile> = ArrayList()
+        project.guessProjectDir()?.let {
+            VfsUtilCore.visitChildrenRecursively(
+                it,
+                object : VirtualFileVisitor<Void?>(NO_FOLLOW_SYMLINKS) {
+                    override fun visitFile(file: VirtualFile): Boolean {
+                        if ((file.isDirectory && isBuildOrBin(file.name)) ||
```

Comment:
need to start consolidating this "known garbage context" exclusion logic

Reply:
yea will add todo there are some similarities between this and /dev and maybe future features